model/Resource/Service/SyncResourceResultIndexer.php

```diff
@@ -15,68 +15,41 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  *
- * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ * Copyright (c) 2021-2022 (original work) Open Assessment Technologies SA;
  */
 
 declare(strict_types=1);
 
 namespace oat\taoAdvancedSearch\model\Resource\Service;
 
+use core_kernel_classes_Resource;
 use oat\generis\model\OntologyAwareTrait;
 use oat\oatbox\service\ConfigurableService;
-use oat\tao\model\search\index\IndexService;
-use oat\tao\model\search\SearchInterface;
-use oat\tao\model\search\SearchProxy;
+use oat\taoAdvancedSearch\model\Index\Listener\ResourceOperationMediator;
 use oat\taoAdvancedSearch\model\Index\Service\IndexerInterface;
-use Throwable;
 
+/**
+ * This service is used by tasks and command line tools.
+ */
 class SyncResourceResultIndexer extends ConfigurableService implements IndexerInterface
 {
     use OntologyAwareTrait;
 
     public function addIndex($resource): void
     {
-        try {
-            $documentBuilder = $this->getIndexerService()->getDocumentBuilder();
-            $this->propagate($documentBuilder);
-
-            $document = $documentBuilder->createDocumentFromResource($resource);
-
-            $totalIndexed = $this->getSearch()->index(
-                [
-                    $document,
-                ]
-            );
-
-            if ($totalIndexed < 1) {
-                $this->logWarning(
-                    sprintf(
-                        'Could not index resource %s (%s): totalIndexed=%d',
-                        $resource->getLabel(),
-                        $resource->getUri(),
-                        $totalIndexed
-                    )
-                );
-            }
-        } catch (Throwable $exception) {
-            $this->logError(
-                sprintf(
-                    'Could not index resource %s (%s). Error: %s',
-                    $resource->getLabel(),
-                    $resource->getUri(),
-                    $exception->getMessage()
-                )
-            );
+        // Use the mediator to forward an event to the AgnosticEventListener
+        // and let it decide how to (re)index the resource based on the
+        // container config
+        //
+        if ($resource instanceof core_kernel_classes_Resource) {
+            $this->getMediator()->handleAddIndex($resource);
```

Comment:
The mediator no longer exists, shall we use the new service `AdvancedSearchDocumentBuilder` and index the code as it was before?

Reply:
This has been already changed to use `AdvancedSearchDocumentBuilder`, marking this thread as solved.